packages/scripts/src/utils/file-utils.ts

```diff
@@ -173,6 +173,12 @@ export interface IContentsEntry {
   modifiedTime: string;
   md5Checksum: string;
   localPath?: string;
+  /** specific path to file when not the same as relativePath, e.g. asset overrides */
+  filePath?: string;
+  /** External URL to display remote assets when running in browser */
+  url?: string;
+  /** Path to file in local device filesystem, populated after asset download */
+  cachedFilepath?: string;
```

Comment:
nit(blocking)
This contents entry is used quite a few other places, if extending for the remote-asset feature probably better to create a new interface entirely (could extend if useful or not)

Also I'm not sure if I fully understand the reason to have filePath or cachedFilePath properties (would they not both  be covered by localPath?)

Reply:
Good point. Instead of adding these extra properties, I've made use of the existing `filePath` property (introduced in [this PR](https://github.com/IDEMSInternational/parenting-app-ui/pull/1874/files#diff-37887cafe4009287e88b297e43b425cb7f04342b1f04689cc6e6310b27b76902)). 

There is a divergence between `IContentsEntry` as defined in `packages/scripts/src/utils/file-utils.ts` (node environment) and `packages/data-models/deployment.model.ts` (runtime environment): the former has the `localPath` property, which refers to the path to the file in the dev's local repo (I think), whereas the latter has the `filePath` property, which gives the path to the file within the app bundle folders, to be read at runtime to get the actual path to the file (this path contains any theme and language segments, unlike the `assetName`).
It's a bit confusing, but as these are different properties I think it makes sense to keep them separate – but then should `IContentsEntry` have both properties in both locations even though each property will only be used in its respective environment?

In terms of this PR, I'm now making use of the `filePath` property to store _either_ the external supabase URL or the internal Android filesystem path (e.g. `file://` etc), depending on the runtime platform. Then the template asset service doesn't need to be aware of the platform and reads the same property regardless. But it is a bit confusing, especially as you might expect `localPath` to store the internal Android filepath, but as mentioned, this refers to the local path on the dev's machine when running content syncs etc.

Reply:
Ah, yeah that's a little on the messy side, no reason to have both `filePath` and `localPath` but don't really like the idea of refactoring as part of this PR. I still expect that longer term we will want to simply extend either of those definitions to add fields to track remote status (e.g. `download_progress`, `download_location` etc.), but fine to keep as-is for now